components/class-m-chart-chartjs.php

```diff
@@ -0,0 +1,328 @@
+<?php
+
+class M_Chart_Chartjs {
+	public $library = 'chartjs';
+	public $value_labels_limit = 15;
+	public $value_labels_div = 10;
+	public $original_labels;
+	public $post;
+	public $post_meta;
+	public $args;
+	public $type_options = array(
+		'line',
+		'column',
+		'bar',
+		'pie',
+	);
+	public $theme_directories;
+	public $colors = array(
+		'#7cb5ec',
+		'#434348',
+		'#90ed7d',
+		'#f7a35c',
+		'#8085e9',
+		'#f15c80',
+		'#e4d354',
+		'#2b908f',
+		'#f45b5b',
+		'#91e8e1',
+	);
+	public $chart_types = array(
+		'column' => 'bar',
+		'bar'    => 'horizontalBar',
+		'pie'    => 'pie',
+		'line'   => 'line',
+	);
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_filter( 'm_chart_image_support', array( $this, 'm_chart_image_support'), 10, 2 );
+	}
+
+	/**
+	 * Get the necessary chart data for a given chart and assign it the right class vars
+	 *
+	 * @param int $post_id the WP ID for the chart post
+	 * @param array $args any args we want to override the defaults for
+	 */
+	public function get_chart_data( $post_id, $args ) {
+		$this->args = wp_parse_args( $args, m_chart()->get_chart_default_args );
+		$this->post = get_post( $post_id );
+		$this->post_meta = m_chart()->get_post_meta( $this->post->ID );
```

Comment:
It might be worth handling the case where the result of `get_post()` is `null`. If it is, `$this->post->ID` will throw an error.